black.py

```diff
@@ -409,7 +409,8 @@ def format_str(src_contents: str, line_length: int) -> FileContent:
     """
     src_node = lib2to3_parse(src_contents)
     dst_contents = ""
-    lines = LineGenerator()
+    future_imports = get_future_imports(src_node)
```

Comment:
We should rather move `py36 = ...` here. If the file is 3.6+ we don't even have to look at the future imports then.

Reply:
Because we want to strip that `u` in 3.6+ code, right?

Reply:
Yes, why keep it for code that has f-strings et al.

Reply:
I just made it do the removal in 3.6-only code, but really of course it should do it in all Python 3-only code. Black currently doesn't have functionality for detecting whether any Python 3 feature is used, and adding that seems a bit too big for this PR. Of the top of my head, we could look for: annotations, class keyword arguments, keyword-only arguments, async/await, print as a function without a future import, extended unpacking.